core/src/consensus/tendermint/mod.rs

```diff
@@ -454,6 +453,16 @@ impl Tendermint {
     pub fn on_imported_proposal(&self, proposal: &Header) {
         let _guard = self.step_change_lock.lock();
 
+        self.votes.throw_out_old(&VoteStep {
```

Comment:
Do we have a proposal for genesis block?
If not, I think it must do an early return if the proposal number is 0 and the below code should be `(proposal.number() - 1) as usize`

Reply:
No, we don't have a proposal for genesis block.
OK. I"ll fix it.